src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java

```diff
@@ -44,26 +46,57 @@ public void loadMethod(FunctionMethodDescriptor descriptor, Map<String, BindingI
 			throws ClassNotFoundException, NoSuchMethodException, IOException {
 		descriptor.validate();
 		addSearchPathsToClassLoader(descriptor);
-		initializeInvocationChainFactory();
+		initializeOneTimeLogics();
 		FunctionDefinition functionDefinition = new FunctionDefinition(descriptor, bindings, classLoaderProvider);
 		this.methods.put(descriptor.getId(), new ImmutablePair<>(descriptor.getName(), functionDefinition));
 	}
 
+	private void initializeOneTimeLogics() {
+		if (!oneTimeLogicInitialized.getAndSet(true)) {
+			initializeInvocationChainFactory();
+			initializeFunctionInstanceInjector();
+		}
+	}
+
 	private void initializeInvocationChainFactory() {
-		if (!invocationChainFactoryInitialized.getAndSet(true)) {
-			ArrayList<Middleware> middlewares = new ArrayList<>();
-			try {
-				//ServiceLoader will use thread context classloader to verify loaded class
-				Thread.currentThread().setContextClassLoader(classLoaderProvider.createClassLoader());
-				for (Middleware middleware : ServiceLoader.load(Middleware.class)) {
-					middlewares.add(middleware);
-					WorkerLogManager.getSystemLogger().info("Load middleware " + middleware.getClass().getSimpleName());
-				}
-			} finally {
-				Thread.currentThread().setContextClassLoader(ClassLoader.getSystemClassLoader());
+		ArrayList<Middleware> middlewares = new ArrayList<>();
+		try {
+			//ServiceLoader will use thread context classloader to verify loaded class
+			Thread.currentThread().setContextClassLoader(classLoaderProvider.createClassLoader());
+			for (Middleware middleware : ServiceLoader.load(Middleware.class)) {
+				middlewares.add(middleware);
+				WorkerLogManager.getSystemLogger().info("Load middleware " + middleware.getClass().getSimpleName());
+			}
+		} finally {
+			Thread.currentThread().setContextClassLoader(ClassLoader.getSystemClassLoader());
+		}
+		middlewares.add(getFunctionExecutionMiddleWare());
+		this.invocationChainFactory = new InvocationChainFactory(middlewares);
+	}
+
+	private void initializeFunctionInstanceInjector() {
+		try {
+			//ServiceLoader will use thread context classloader to verify loaded class
+			Thread.currentThread().setContextClassLoader(classLoaderProvider.createClassLoader());
+			Iterator<FunctionInstanceInjector> iterator = ServiceLoader.load(FunctionInstanceInjector.class).iterator();
+			if (iterator.hasNext()) {
+				this.functionInstanceInjector = iterator.next();
+				WorkerLogManager.getSystemLogger().info("Load function instance injector: " + this.functionInstanceInjector.getClass().getName());
```

Comment:
```suggestion
				WorkerLogManager.getSystemLogger().info("Load function instance injector: {0}", this.functionInstanceInjector.getClass().getName());
```

Reply:
Hi @trask , I only found two `info` methods in `java.utli.logging`, [one ](https://docs.oracle.com/javase/8/docs/api/java/util/logging/Logger.html#info-java.lang.String-)takes one `String` parameter, the [other one](https://docs.oracle.com/javase/8/docs/api/java/util/logging/Logger.html#info-java.util.function.Supplier-) takes one `Supplier<String>` as parameter. Seems `java.utli.logging` doesn't support formatting log like above.